pkg/policy/l4.go

```diff
@@ -220,41 +225,47 @@ func (l7 L7DataMap) addRulesForEndpoints(rules api.L7Rules, endpoints []api.Endp
 			l7[epsel] = rules
 		}
 	} else {
-		// If there are no explicit fromEps, have a 'special' wildcard endpoint.
-		l7[api.WildcardEndpointSelector] = rules
+		panic("Must have at least the wildcard selector")
 	}
 }
 
-// CreateL4Filter creates a filter for L4 policy that applies to the specified
+// delete releases the references held in the L4Filter and must be called before
+// the filter is left to be garbage collected!
+func (l4 *L4Filter) delete(selectorCache *SelectorCache) {
+	selectorCache.RemoveSelectors(l4, l4.CachedSelectors)
+	l4.CachedSelectors = nil
+	l4.L7RulesPerEp = nil
+}
+
+// createL4Filter creates a filter for L4 policy that applies to the specified
 // endpoints and port/protocol, with reference to the original rules that the
 // filter is derived from. This filter may be associated with a series of L7
 // rules via the `rule` parameter.
-func CreateL4Filter(peerEndpoints api.EndpointSelectorSlice, rule api.PortRule, port api.PortProtocol,
-	protocol api.L4Proto, ruleLabels labels.LabelArray, ingress bool) L4Filter {
+func createL4Filter(peerEndpoints api.EndpointSelectorSlice, rule api.PortRule, port api.PortProtocol,
```

Comment:
This is outside the scope of this PR, but we should strongly consider changing the parameters of these functions to be some type of utility struct containing all the parameters wrapped in a single type, because it keeps growing.